bookmarks/tests.py

```diff
@@ -0,0 +1,20 @@
+import pytest
+from django.db.models.query import QuerySet
+from bookmarks.models import Bookmark
+
+
+@pytest.mark.django_db
+class TestBookmark:
+    def test_course_feed(cls):
+        '''
+        this function tests the bookmarks displayed on the course page
+        '''
+        out = Bookmark.course_feed()
+        assert isinstance(out, QuerySet)
+        assert all(isinstance(b, Bookmark) for b in out)
+        assert list(out.values_list('user', 'course', 'url', 'urlname')) == [
+            (2, 2, 'https://www.facebook.com', 'Facebook'),
+            (1, 2, 'https://www.facebook.com', 'Facebook'),
+            (1, 1, 'https://www.ynet.co.il', 'Ynet'),
+            (1, 1, 'https://www.google.com', 'Google'),
+        ]
```

Comment:
This will work as an initial test based on the assumption that the migration data is the only data in the system. However, a test module shouldn't assume that specific data is in the system unless that is part of the production system that it is testing. For the test module to be stand-alone, the data that you are verifying should be added in the test module itself, and should use a filter so you get the specific data you are looking for instead of using .all
For example, adding a new bookmark in the test script and verifying that it is first in the Bookmark list.

Reply:
What do you mean by ' the data that you are verifying should be added in the test module itself'? The data for the tests are already in the '0002_test_data' file as Barak showed us in class. As you can see in bookmarks.models I have a 'last-view-date' field on Bookmark class that supposed to set the order of the bookmarks list from the recently used/watched bookmark to the least used one. As for you example- I did a similar idea and used a classmethod that arranges the bookmarks data by it's last-view-date.

Reply:
After discussing this with the Red Hat team, this is a valid method of doing it for the course.